### PR TITLE
feat(front-end): send resizes, and fullscreen editor

### PR DESCRIPTION
**Pull request by gwimm**

### CODE REVIEW

**Review by elkowar — CHANGES_REQUESTED** [review]
Could use a little cleanup, but generally this is a lot cleaner than it was before! (And it works, too)

File: rekuho/src/lib/rpc.ts [hunk 1]
```diff
@@ -29,36 +27,58 @@ export class Session {
...
+      this.send({ method: "mouse_input", params: { view_id, position } })
+    }
+  }
+
+  handleUpdateView(args: {
+    height: number // line count
+    width: number // column count
+    first_line: number
+    first_col: number
+  }) {
```
[the remark this answers is not in the document]
Comment (gwimm) [reply rekuho/src/lib/rpc.ts:49]: Fixed.

File: rekuho/src/lib/font.ts [hunk 2]
```diff
@@ -1,3 +1,9 @@
+import type { Font } from "./theme"
+
+export const toString = (font: Font): string => {
```
[the remark this answers is not in the document]
Comment (gwimm) [reply rekuho/src/lib/font.ts:3]: it's `toString` in the `lib/font.ts` file. you're asking of something absolutely horrendeuous, when you insist on `col`.

[the remark this answers is not in the document]
Comment (gwimm) [reply rekuho/src/lib/font.ts:3]: fixed

File: rekuho/src/lib/Portion.svelte [hunk 3]
```diff
@@ -4,154 +4,107 @@
...
-  /*
-  const mousedown = (ev: MouseEvent) => {
-    const current = pxToPortionPosition([ev.pageX, ev.pageY])
-    selection = current
-    cursorUpdate(0, (_) => current)
-    input.focus()
+  $: if (container) {
+    const fontMetrics = fontMeasure(container, fontToString(theme.font))
+    lineHeight = fontMetrics.actualHeight || 0
+    columnWidth = fontMetrics.width || 0
```
[the remark this answers is not in the document]
Comment (gwimm) [reply rekuho/src/lib/Portion.svelte:54]: Now semantically correct.

File: rekuho/src/lib/Portion.svelte [hunk 4]
```diff
@@ -4,154 +4,107 @@
...
     mousedown: CaretPosition
   }>()
 
   const emitKeyboardInput = (input: KeyInput) => dispatch("keyinput", input)
   const onMouseClicked = (pos: CaretPosition) => dispatch("mousedown", pos)
 
-  $: view_rect = container && container.getBoundingClientRect()
+  ////////////////////////////////////////////////////////////////////////////////
+
+  $: viewRect = container && container.getBoundingClientRect()
```
[the remark this answers is not in the document]
Comment (gwimm) [reply rekuho/src/lib/Portion.svelte:37]: No

File: rekuho/src/lib/font.ts [hunk 5]
```diff
@@ -28,3 +32,48 @@ export const measure = (
...
+    pre.appendChild(document.createTextNode("x"))
+    pre.appendChild(elt("br"))
+  }
+  pre.appendChild(document.createTextNode("x"))
+
+  removeChildrenAndAdd(element, pre)
+
+  const height = pre.offsetHeight / 50
+  return height || 1
+}
```
[the remark this answers is not in the document]
Comment (gwimm) [reply rekuho/src/lib/font.ts:79]: fixed

**Review by elkowar — APPROVED** [review]
Looks pretty good now!

**Pull request merged**